cli/azd/cmd/container.go

```diff
@@ -496,8 +496,11 @@ func registerCommonDependencies(container *ioc.NestedContainer) {
 		// Project Configuration
 		var projectConfig *devcenter.Config
```

Comment:
I'm wondering if the code should have "devcenter" here or if it should be "platform" or "platformType" to be more generic.

Reply:
I agree that there is more here we can do.  We don't have additional examples for what a platform provider is outside of devcenter.  I could see the introduction of some sort of `PlatformProvider` component that basically just sets or overrides configuration.

We have the user facing bits in place to make this easily extensible and will think more on how we handle this within our code.  I also don't want to have `devcenter` type things littered over the code base.